### PR TITLE
Remove memoizing static URLs

### LINKED ISSUE #350: Staticfiles URL Caching issue
When using staticfiles backends such as S3 that calculate signatures on the fly which are only valid for a limited time, this method causes issues, because since the signature is only calculated once and then cached, after it expires the file will be unreachable.

https://github.com/makinacorpus/django-leaflet/blob/5d35bda1833a3cdcbc553c2b16fce9c9ece26d94/leaflet/__init__.py#L171

I feel like dropping the caching would not negatively impact the performance all too much, as the staticfiles performs quite well.

### PR DESCRIPTION
**Pull request by Schlepptop**
Fixes #350

### CODE REVIEW

File: leaflet/utils.py [hunk 1]
```diff
@@ -68,13 +44,6 @@ def __add__(self, iterable):
         lazy_list.extend(iterable)
         return lazy_list
 
-    @classmethod
-    def _resolve_lazy_item(cls, item):
```
Comment (Gagaro) [inline leaflet/utils.py:72]: Why did you remove that?

**Review by Gagaro — COMMENTED** [review]
I agree on the memoize. Why did you remove the `_resolve_lazy_item` though?

File: leaflet/utils.py (hunk 1 above)
Comment (Schlepptop) [reply leaflet/utils.py:72]: It wasn't needed anywhere else as far as I can tell, so I just dropped it. Can revert that ofc.

Comment (Gagaro) [reply leaflet/utils.py:72]: It is used in `__iter__` / `__getitem__` just above. And as far as I can tell it as nothing to do with the memoize. I'd rather let `ListWithLazyItems` as it was before to avoid any bug.

Comment (Schlepptop) [reply leaflet/utils.py:72]: We can't use the default Django `lazy` function then because the `__proxy__` class doesn't define a `__call__` method which is used in the old `_resolve_lazy_items`.

Comment (Schlepptop) [reply leaflet/utils.py]: We actually don't even need the custom lazy stuff in `utils.py` if we use the standard Django `lazy` function, so I just reverted those changes and modified `__init__.py` accordingly.

**Comment by Gagaro** [comment]
Thanks again for the PR :+1: .
